### ISSUE TITLE
Content gets heavily reordered after "flatten"

### ISSUE THREAD

**Issue by eliasbalasis** [mojohaus/flatten-maven-plugin#400]
Support for [Maven CI Friendly Versions](https://maven.apache.org/maven-ci-friendly.html) is great
but the installed/deployed content is completely different to the original after using `flatten:flatten`.

Properties are mixed and original order/grouping is indecipherable, making hard to track large numbers of inherited properties.

Can the order/grouping of original content be preserved?

**Comment by stechio** [comment]
You're absolutely right! That's a problem affecting Maven's Effective POM serialization, which presumably is delegated by flatten-maven-plugin to generate the flattened POM.

Really weird behavior, considering that POM 4.0 schema ([http://maven.apache.org/POM/4.0.0](https://maven.apache.org/xsd/maven-4.0.0.xsd)) declares properties as an ordered list (so                        its order should be honored on transformation, since the Effective POM itself belongs to POM 4.0 schema):

```xsd
<xs:element minOccurs="0" name="properties">
  <xs:annotation>
    <xs:documentation source="version">4.0.0+</xs:documentation>
    <xs:documentation source="description">
    Properties that can be used throughout the POM as a substitution, and are used as filters in resources if enabled. The format is <code>&lt;name&gt;value&lt;/name&gt;</code>.
    </xs:documentation>
  </xs:annotation>
  <xs:complexType>
    <xs:sequence>
      <xs:any minOccurs="0" maxOccurs="unbounded" processContents="skip"/>
    </xs:sequence>
  </xs:complexType>
</xs:element>
```
